File: proxy.go
package main

import (
    "bytes"
    "encoding/json"
    "io"
    "net/http"
    "os"
)

func main() {
    apiKey := os.Getenv("VITE_DEEPSEEK_API_KEY")
    if apiKey == "" {
        panic("API key not found in environment")
    }

    http.HandleFunc("/api/chat", func(w http.ResponseWriter, r *http.Request) {
        // Handle CORS
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, OPTIONS")
        w.Header().Set("Access-Control-Allow-Headers", "Content-Type, Authorization")

        if r.Method == "OPTIONS" {
            w.WriteHeader(http.StatusOK)
            return
        }

        // Read request body
        body, err := io.ReadAll(r.Body)
        if err != nil {
            http.Error(w, "Bad request", http.StatusBadRequest)
            return
        }

        // Forward to Taobao API
        req, err := http.NewRequest("POST", "https://tb.api.mkeai.com/v1/chat/completions", bytes.NewReader(body))
        if err != nil {
            http.Error(w, err.Error(), http.StatusInternalServerError)
            return
        }

        req.Header.Set("Content-Type", "application/json")
        req.Header.Set("Authorization", "Bearer "+apiKey)

        client := &http.Client{}
        resp, err := client.Do(req)
        if err != nil {
            http.Error(w, err.Error(), http.StatusBadGateway)
            return
        }
        defer resp.Body.Close()

        // Copy response
        respBody, err := io.ReadAll(resp.Body)
        if err != nil {
            http.Error(w, err.Error(), http.StatusInternalServerError)
            return
        }

        w.Header().Set("Content-Type", "application/json")
        w.WriteHeader(resp.StatusCode)
        w.Write(respBody)
    })

    println("Starting proxy server on :3001")
    println("Forwarding API requests to Taobao DeepSeek")
    http.ListenAndServe(":3001", nil)
}

File: proxy.py
from http.server import HTTPServer, BaseHTTPRequestHandler
import requests
import json
import os
from dotenv import load_dotenv

load_dotenv('.env.local')

API_KEY = os.getenv('VITE_DEEPSEEK_API_KEY')
API_URL = 'https://tb.api.mkeai.com/v1/chat/completions'

class ProxyHandler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        self.end_headers()
    
    def do_POST(self):
        if self.path == '/api/chat':
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            
            try:
                # Forward to Taobao API
                headers = {
                    'Content-Type': 'application/json',
                    'Authorization': f'Bearer {API_KEY}'
                }
                
                response = requests.post(API_URL, headers=headers, data=post_data, timeout=30)
                
                self.send_response(response.status_code)
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Content-Type', 'application/json')
                self.end_headers()
                self.wfile.write(response.content)
                
            except Exception as e:
                self.send_response(500)
                self.send_header('Content-Type', 'application/json')
                self.end_headers()
                error_msg = json.dumps({'error': str(e)})
                self.wfile.write(error_msg.encode())
        else:
            self.send_response(404)
            self.end_headers()

if __name__ == '__main__':
    server = HTTPServer(('localhost', 3001), ProxyHandler)
    print('🚀 Proxy server running on http://localhost:3001')
    print('📡 Forwarding to Taobao API...')
    server.serve_forever()
